tests/test_security_hardening.py

```
@@ -20,6 +22,25 @@ def build() -> TaskOrchestrator:
+def test_external_bearer_verifier_is_fail_closed_and_scoped() -> None:
+    seen: list[tuple[str, str]] = []
+
+    def verify(token: str, scope: str) -> bool:
+        seen.append((token, scope))
+        return token == "keyverse-token" and scope == "inference"
+
+    security = SecurityConfig(bearer_verifier=verify)
+    security.authorize({"authorization": "Bearer keyverse-token"}, "inference", "127.0.0.1")
+    try:
+        security.authorize({"authorization": "Bearer keyverse-token"}, "admin", "127.0.0.1")
+    except Exception as exc:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)